enterprise/trackedusers/data_collector.go

```diff
@@ -0,0 +1,192 @@
+package trackedusers
+
+//go:generate mockgen -destination=./mocks/mock_data_collector.go -package=mockdatacollector github.com/rudderlabs/rudder-server/enterprise/trackedusers DataCollector
+
+import (
+	"context"
+	"database/sql"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	migrator "github.com/rudderlabs/rudder-server/services/sql-migrator"
+
+	obskit "github.com/rudderlabs/rudder-observability-kit/go/labels"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+
+	"github.com/lib/pq"
+
+	"github.com/spaolacci/murmur3"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/segmentio/go-hll"
+
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	txn "github.com/rudderlabs/rudder-server/utils/tx"
+)
+
+const (
+	idTypeUserID                       = "userID"
+	idTypeAnonymousID                  = "anonymousID"
+	idTypeUserIDAnonymousIDCombination = "userIDAnonymousIDCombination"
+
+	// changing this will be non backwards compatible
+	murmurSeed = 123
+)
+
+// DataCollector is interface to collect data from jobs
+type DataCollector interface {
+	CollectData(ctx context.Context, jobs []*jobsdb.JobT, tx *txn.Tx) error
+}
```

Comment:
[subjective]
```suggestion
type UserTracker interface {
	Track(ctx context.Context, tx *txn.Tx,  jobs []*jobsdb.JobT) error
}
```

* `Data` is a bit generic, I do prefer the more applicable `User` term.
* Track sounds better than Collect, as we do keep any of the jobs rather use them to infer some data

Reply:
Track feels like we are tracking user activity
how about ?
```go
type UsersReporter interface {
	ReportUsers(ctx context.Context, reports []*UsersReport, tx *txn.Tx) error
}
```